torch/optim/adam.py

```diff
@@ -37,14 +39,18 @@ def __init__(self, params, lr=1e-3, betas=(0.9, 0.999), eps=1e-8,
                 raise RuntimeError("`fused` does not support `differentiable`")
             self._step_supports_amp_scaling = True
             # TODO(crcrpar): [low prec params & their higher prec copy]
-            # Suppor AMP with FP16/BF16 model params which would need
+            # Support AMP with FP16/BF16 model params which would need
             # higher prec copy of params to do update math in higher prec to
             # alleviate the loss of information.
+            if not all(_check_same_device(pg['params']) for pg in self.param_groups):
+                raise RuntimeError("`fused=True` requires all the params to be on same device.")
```

Comment:
This isn't true. We group by device and dtype later on when calling torch.fused_adam so this requirement can be relaxed.